### PR TITLE
ARROW-12643: [Governance] Added experimental repos guidelines.

### PR DESCRIPTION
**Pull request by jorgecarleitao**

### CODE REVIEW

File: docs/source/developers/experimental_repos.rst
Comment (andygrove) [inline docs/source/developers/experimental_repos.rst:41]:
```suggestion
* Any decision to make the experimental repo official in any way, whether by merging or migrating, *must* be discussed and voted on in the mailing list.
```

**Comment by jorgecarleitao** [comment]
@wesm @emkornfield @andygrove , is there anything we need to do here?

**Comment by emkornfield** [comment]
Sorry for the delay. LGTM to me.  I'll wait for feedback from others before merging.

**Review by andygrove — APPROVED** [review]
LGTM!

File: docs/source/developers/experimental_repos.rst
Comment (andygrove) [inline docs/source/developers/experimental_repos.rst:52]: We should also require CI checks for ASF header in files

**Comment by jorgecarleitao** [comment]
if no-one opposes, I plan to merge this tomorrow.

**Pull request closed without merging**
